client/task_runner.go

```diff
@@ -384,7 +396,23 @@ func (r *TaskRunner) run() {
 				if err := r.handleUpdate(update); err != nil {
 					r.logger.Printf("[ERR] client: update to task %q failed: %v", r.task.Name, err)
 				}
+			case err := <-r.vaultRenewalCh:
+				if err == nil {
+					// Only handle once.
```

Comment:
Hm, I don't really understand this comment. It seems like you're just spinning on this particular chan until it's non-nil (or another case is available)?